lib/pure/uri.nim

```diff
@@ -8,6 +8,30 @@
 #
 
 ## This module implements URI parsing as specified by RFC 3986.
+##
+## A Uniform Resource Identifier (URI) provides a simple and extensible
+## means for identifying a resource. A URI can be further classified
+## as a locator, a name, or both. The term “Uniform Resource Locator”
+## (URL) refers to the subset of URIs.
+##
+## .. code-block::
```

Comment:
Please consider putting these examples behind a heading which describes the example. Otherwise we are setting a precedent for having multiple unlabelled examples and the users will have no way to quickly look through them.

Perhaps your intention is to just show two highlights though, in which case it might be good to make that explicit too?

Reply:
> I like to include a heading - it gives a more structured look.

I've used headings where there are multiple parts of a module I would like to show, e.g. see [tables](https://nim-lang.github.io/Nim/tables.html).  
If there is only one example which showcases the whole module, then I didn't introduce "Basic usage" heading, e.g. see [strutils](https://nim-lang.github.io/Nim/strutils.html). 

Here, where you would have multiple examples, showing different things, I would do it the way you did it now.


> what about the other modules? The same structure across modules would be preferred.

I use "Basic usage", you have "Examples". If you would like to have consistency, I would prefer if you would change it to "Basic usage", as I already used that in multiple modules (some remain to be published).

Reply:
Changed to "Basic usage".